### PR TITLE
Refactoring classes and properties

### PR DESCRIPTION
**Pull request by kmruehl**
Refactoring WEC-Sim Classes and Properties based on https://github.com/WEC-Sim/WEC-Sim/projects/58#card-64210391 and discussions with the WEC-Sim team. 

**Objectives:**
- clear and intuitive variable naming using camelCase
- consistent properties across classes, e.g. `pto.loc` versus `mooring.ref`
- convert properties for a single feature to property structures, e.g. `struct(body.morison)` 
- move properties to the appropriate class, e.g.  `simu.yaw` to `body.yaw`
- updating WEC-Sim library accordingly

**Wave Class**
- [x] `waves.etaImport` to `waves.waveImport` to `waves.elevationImport`
- [x]  `waves.spectrumDataFile` to `waves.waveSpectrumFile` to `waves.spectrumFile`
- [x] `waves.etaDataFile` to `waves.waveElevationFile`to `waves.elevationFile`
- [x] `waves.plotEta` to `waves.plotElevation`
- [x] `waves.waveDir` to `waves.waveDirection` to `waves.direction`
- [x] `waves.waveSpread` to `waves.spread`
- [x] `beta` to `theta` to match online documentation
- [x] `waves.deepWaterWave` to `waves.deepWater`
- [x] convert `waves.current` properties to structure, similar to morison element, 
  - `waves.currentOption` to `waves.current.option`
  - `waves.currentDepth` to `waves.current.depth`
  - `waves.currentDirection` to `waves.current.direction`
  - `waves.currentSpeed` to `waves.current.speed`
- [x] convert properties related to BEM data to `waves.bem` structure
  - `waves.numFreq` to `waves.freqNum` to `waves.bem.count`
  - `waves.freqDisc` to `waves.bem.option`
  - `waves.freqRange` to `wave.bem.range`
- [x] add checks for waveClass structures
- [x] convert `waves.marker` properties to structure
   - [x] `waves.markerLoc` to `waves.marker.loc` to `waves.marker.location`
   - [x] `waves.markerSize` to `waves.marker.size`
   - [x] `waves.markerStyle` to `waves.marker.style` 
   - [x] convert `waves.marker` properties to structure in Frames Library

**Simulation Class**
- [x] `simu.writetxt` to `simu.writeText`
- [x] `waves.waveStatisticsDataLoad` to `simu.mcrExcelFile`  (formerly Wave Class)
- [x] `simu.mcrCaseFile` to `simu.mcrMatFile`
- [x] `simu.dtME` to `simu.morisonDt`
- [x] `simu.CTTime` to `simu.cicTime`
- [x] `simu.dtCI` to `simu.cicDt`
- [x] `simu.CITime` to `simu.cicEndTime` 
- [x] `simu.CIkt` to `simu.cicLength`
- [x] `simu.ssCalc` to `simu.stateSpace`
- [x] `simu.setupSimu` to `simu.simuSetup` to `simu.setup`
- [x] convert `simu.paraview` properties to structure, similar to morison element, note: test that sub-properties are tested 
    - `simu.dtParaview` to `simu.paraviewDt` to `simu.paraview.dt`
    - `simu.StartTimeParaview` to  `simu.paraviewStartTime` to  `simu.paraview.startTime`
    - `simu.EndTimeParaview` to `simu.paraviewEndTime` to `simu.paraview.endTime`
    - `simu.pathParaviewVideo` to `simu.paraviewDirectory` to `simu.paraview.path`
- [x] `simu.numWecBodies` to `simu.body.numHydroBodies`
- [x] `simu.pressureDis` to `simu.pressure`
- [x] `simu.simulationDate` to `simu.date`
- [x] add checks for simulationClass structures
- [x] `simu.adjMassWeightFun` to `simu.adjMassFactor`
- [x] `simu.autoRateTranBlk` to `simu.rateTransition`
- [x] `simu.zeroCrossCont` to `simu.zeroCross`
- [x] `simu.numIntMidTimeSteps` removed

**Body Class**
- [x] `body.dof_start` to ``body.dofStart``
- [x] `body.dof_end` to ``body.dofEnd``
- [x] `body.dof_gbm` to ``body.dofGBM``
- [x] `body.nlHydro` to ``body.nonlinearHydro``
- [x] `body.lenJ` to ``body.dofCoupled``
- [x] `body.meanDriftForce` to ``body.meanDrift``
- [x] `body.bodyparaview` to ``body.paraviewBody`` to ``body.paraview``
- [x] `body.flexHydroBody` to ``body.flexBody`` to ``body.flex``
- [x] `body.nhBody` to ``body.nonHydroBody`` to ``body.nonHydro``
- [x] `body.hydroDataBodyNum` to ``body.hydroBodyTotal`` to `body.hydroTotal` **NOTE** Do we even use this?
- [x] `body.viscDrag` to `body.viscousDrag` to `body.quadDrag`
- [x] `body.userDefinedExcIRF` to `body.excitationIRF`
- [x] add checks for bodyClass structures
- [x] move `simu.yaw` and `simu.yawThresh` to bodyClass 
   - [x] `simu.yaw` to `body.yaw.option`
   - [x] `simu.yawThres` to `body.yaw.threshold`
- [x] `body.bodyNumber` to `body.number`
- [x] `body.bodyTotal` to `body.total`
- [x] `body.bodyGeometry` to `body.geometry`

**Cable Class**
- [x] `cable.c` to `cable.damping`
- [x] `cable.k` to `cable.stiffness`
- [x] `cable.viscDrag` to `cable.viscousDrag` to `cable.quadDrag` 
- [x] `cable.cableNum` to `cable.number`
- [x] `cable.loc` to `cable.location`

**Constraint Class**
- [x] `constraint.loc` to `constraint.location`
- [x] `constraint.constraintNum` to `constraint.number`

**PTO Class**
- [x] `pto.loc` to `pto.location`
- [x] `pto.c` to `pto.damping`
- [x] `pto.k` to `pto.stiffness`
- [x] `pto.ptoNum` to `pto.number`

**Mooring Class**
- [x] `mooring.ref` to `mooring.loc` to `mooring.location`
- [x] `mooring.loc` to `mooring.orientation` (internal)
- [x] `mooring.mooringNum` to `mooring.number`

**Doc**
- [ ] add dev documentation for style guide, look into naming convention, e.g. pep 8

**Notes**
- [x] Are the property structure fields checked? No, properties are automatically checked, but not the structure fields. For example there's an error if `waves.bems` is specified in the input file, but not for `waves.bem.options`. This can be addressed by adding some additional checks to the `checkinputs` method of each class with a structure.
- [ ] how does b2b impact migration of passive yaw to bodyClass?
- [ ] resolve `TestPassiveYawRegression` test failure
- [ ] resolve `runFromSimTest` test issues, fails on first run but not second

### CODE REVIEW

**Comment by kmruehl** [comment]
@akeeste The tests right now are passing because I did not update the run-from-simulink parameters. I'll be out for the working meeting, but if you'd like to work on this PR to update the new variables in the run-from-simulink cases that would be a huge help. On that note... is here a best workflow to find/replace variable names for run-from-simulink?

**Comment by akeeste** [comment]
@kmruehl A few thoughts/suggestions:

1. I like how the new format for variables is "featureParameter". For example renaming the simu parameters by feature first, helps keep them organized and grouped together. Maybe we lump all parameters for a given feature into the same struct? (like how ``bodyClass.morisonElement`` is setup). We could create ``paraview``, ``cic`` structs for the simulation class that then hold time, end time, time step, etc.
2. I like that all the body descriptions are the same format now (i.e. ``typeBody`` format)
3. I don't think parameters or variables should repeat the name of the class. This makes them longer with adding information. For example ``waves.waveType`` and ``simu.simuSetup()`` are redundant, I prefer just ``type`` and ``setup()``. Similarly we can take "body" out of any bodyClass parameter without loss of information.
4. I think functions are more clear in a format like "actionObject", e.g. ``setupSimu``, ``setInitDisp``, ``setupRhoDensity`` over ``simuSetup``, ``initDispSet``, ``rhoDensitySetup``. It seems most but not all of our class methods are like this already.
5. ``waves.waveNumber()`` should've been removed in #790... I think this got lost when we merged PRs. I confirmed that only the BEMIO function is being called now, and ``waves.waveNumber`` should be removed.
6. compilationTest became the application tests last release and passiveYawTest was moved to that application. They should not be in the WEC-Sim repo now

Also, its not really renaming related, but can we make yaw a bodyClass parameter instead of simulation wide?

**Comment by nathanmtom** [comment]
@kmruehl Thanks for working on the variable clean up! I believe having a more consistent naming structure will help improve the users ability to discern what each variable represents. 

My thoughts/suggestions:
1. For waves.numFreq, the variable was changed to waves.freqNum while the other variable changes had the words written out completely. Therefore, I'd request writing as waves.frequencyNumber and although it is a longer variable I think it removes uncertainty in what the variable is supposed to define. 
2. To @akeeste point about moving the `yaw` to the body class, I think this makes sense since we moved both `nonlinearHydro` and 'morison' to the body class we should do this also for yaw so bodies that do not need this functionality can turn it off. 
3. To @akeeste point about repeating the class name, agree with the comment and make we can remove any mention of the class name in the structure field variable descriptions.

Happy to discuss these options during the next WEC-Sim working meeting and assist however you would like.

**Comment by kmruehl** [comment]
@H0R5E any ideas why the test are no longer running on this PR?

**Comment by H0R5E** [comment]
> @H0R5E any ideas why the test are no longer running on this PR?

@kmruehl, the CI won't run if there is a potential merge conflict. You'll have to resolve the conflict first.

**Comment by kmruehl** [comment]
@akeeste and @nathanmtom I'm sure I'll find bugs with this and need to make a few more commits but this PR is ready for a review. Please refer to my *Notes* with unresolved issues

File: .github/workflows/dispatch.yml [hunk 1]
```diff
@@ -1,18 +0,0 @@
...
-    name: Repository Dispatch
-    runs-on: ubuntu-latest
-    if: github.repository_owner == 'WEC-Sim'
-    steps:
-    - uses: peter-evans/repository-dispatch@v1
-      with:
-        token: ${{ secrets.REPO_ACCESS_TOKEN }}
-        repository: WEC-Sim/WEC-Sim_Applications
-        event-type: wecsim-${{ github.ref_name }}
-        client-payload: '{"sha": "${{ github.sha }}"}'
```
Comment (H0R5E) [inline .github/workflows/dispatch.yml:18]: @kmruehl, don't delete this file

**Review by H0R5E — CHANGES_REQUESTED** [review]

File: .github/workflows/dispatch.yml (hunk 1 above)
Comment (kmruehl) [reply .github/workflows/dispatch.yml]: resolved with https://github.com/WEC-Sim/WEC-Sim/pull/803/commits/f9946c228e1e0dc14371396663e6189992082423

**Comment by kmruehl** [comment]
@akeeste @H0R5E @nathanmtom  I'm going to go ahead and merge this PR and then re-open a new one for review.

File: source/functions/simulink/model/irregWaveMorison.m [hunk 2]
```diff
@@ -1,4 +1,4 @@
-function f = irregWaveMorison(z,r,Vel,Accel,Disp,Area,Cd,Vol,Ca,Time,rho,waveDir,waterDepth,k,w,A,rampTime,g,randPhase,dw,currentSpeed,currentDirection,currentDepth,currentOption,simuMorison)
+function f = irregWaveMorison(z,r,Vel,Accel,Disp,Area,Cd,Vol,Ca,Time,rho,direction,waterDepth,k,w,A,rampTime,g,randPhase,dw,current.speed,current.direction,current.depth,current.option,simuMorison)
```
Comment (akeeste) [inline source/functions/simulink/model/irregWaveMorison.m:1]: passing variables as struct?

File: source/functions/simulink/model/regWaveMorison.m [hunk 3]
```diff
@@ -1,4 +1,4 @@
-function f = regWaveMorison(z,r,Vel,Accel,Disp,Area,Cd,Vol,Ca,Time,rho,waveDir,waterDepth,k,w,A,rampTime,g,currentSpeed,currentDirection,currentDepth,currentOption,simuMorison)
+function f = regWaveMorison(z,r,Vel,Accel,Disp,Area,Cd,Vol,Ca,Time,rho,direction,waterDepth,k,w,A,rampTime,g,current.speed,current.direction,current.depth,current.option,simuMorison)
```
Comment (akeeste) [inline source/functions/simulink/model/regWaveMorison.m:1]: passing variables as struct?

File: source/objects/responseClass.m [hunk 4]
```diff
@@ -511,19 +491,16 @@ function saveViz(obj,simu,body,waves,options)
...
                     hold off
                 end
-            end
-            
+            end            
             % Close video file
             if options.saveSetting == 0
                 close(video); 
-            end
-
```
Comment (akeeste) [inline source/objects/responseClass.m:523]: Removing all of these blank lines makes this much harder to read

File: tests/runFromSimTest.m [hunk 5]
```diff
@@ -119,31 +99,26 @@ function removeInputFiles(testCase)
...
-            sim(simFile, [], simset('SrcWorkspace','current'));
-            
+            sim(simFile, [], simset('SrcWorkspace','current'));            
             close_system(simFile,0);
             bdclose('all')
             clear body constraint output pto simu waves
         end
         
         function fromSimInput(testCase)
             % Run WEC-Sim from Simulink with input file
```
Comment (akeeste) [inline tests/runFromSimTest.m:113]: same with the empty lines here, I think its much harder to read and pick out sections of the code

File: source/wecSimMCR.m [hunk 6]
```diff
@@ -64,20 +64,20 @@
     
     if exist('pto','var')
         for n=1:size(pto,2)
-            if (length(pto(n).c)>1 || length(pto(n).k)>1)
+            if (length(pto(n).c)>1 || length(pto(n).stiffness)>1)
```
Comment (akeeste) [inline source/wecSimMCR.m:67]: pto(n).damping

File: source/wecSimMCR.m [hunk 7]
```diff
@@ -64,20 +64,20 @@
...
                 name = sprintf('pto(%i).c', n);
                 mcr.header{numConditions-1} = name;
-                name = sprintf('pto(%i).k', n);
+                name = sprintf('pto(%i).stiffness', n);
                 mcr.header{numConditions  } = name;
                 
                 len = length(mcr.cases(:,1)); kkk = 0;
-                for l2=1:length(pto(n).k)
+                for l2=1:length(pto(n).stiffness)
                     for l1=1:length(pto(n).c)
```
Comment (akeeste) [inline source/wecSimMCR.m:76]: pto(n).damping

File: source/wecSimMCR.m [hunk 8]
```diff
@@ -64,20 +64,20 @@
...
                         kkk=kkk+1;
                         mcr.cases(len*(kkk-1)+1:len*(kkk-1)+len,1:numConditions-2) = mcr.cases(1:len,1:numConditions-2);
                         mcr.cases(len*(kkk-1)+1:len*(kkk-1)+len,  numConditions-1) = pto(n).c(l1);
```
Comment (akeeste) [inline source/wecSimMCR.m:79]: pto(n).damping

File: source/objects/waveClass.m [hunk 9]
```diff
@@ -706,15 +703,15 @@ function waveElevUser(obj,rampTime,dt,maxIt,data,time)
...
                 warning('Cannot use wave gauges or visualization markers with eta import. Gauges and markers removed.');
-                obj.markerLoc = [];
+                obj.marker.location = [];
             end
         end
         
         function printWaveSpectrumType(obj)
             % Lists the wave spectrum type
-            % Used by listInfo
+            % Used by  :meth:`waveClass.listInfo`.
```
Comment (akeeste) [inline source/objects/waveClass.m:714]: should this be ``:math:`waveCLass.listInfo`.``?

File: source/objects/bodyClass.m [hunk 10]
```diff
@@ -303,70 +368,49 @@ function setInitDisp(obj, relCoord, axisAngleList, addLinDisp)
...
-            
-        end
-        
-        function listInfo(obj)
-            % This method prints body information to the MATLAB Command Window.
-            fprintf('\n\t***** Body Number %G, Name: %s *****\n',obj.hydroData.properties.body_number,obj.hydroData.properties.name)
-            fprintf('\tBody CG                          (m) = [%G,%G,%G]\n',obj.hydroData.properties.cg)
-            fprintf('\tBody Mass                       (kg) = %G \n',obj.mass);
-            fprintf('\tBody Diagonal MOI              (kgm2)= [%G,%G,%G]\n',obj.momOfInertia)
+            obj.initDisp.initAngularDispAngle = netAngle;
```
Comment (akeeste) [inline source/objects/bodyClass.m:384]: personal preference, but this is harder to read

File: source/objects/bodyClass.m [hunk 11]
```diff
@@ -576,53 +593,53 @@ function constAddedMassAndDamping(obj,w,CIkt,rho,B2B)
...
             else
-                obj.hydroForce.fAddedMass=obj.hydroData.hydro_coeffs.added_mass.inf_freq(:,obj.dof_start:obj.dof_end) .*rho;
+                obj.hydroForce.fAddedMass=obj.hydroData.hydro_coeffs.added_mass.inf_freq(:,obj.dofStart:obj.dofEnd) .*rho;
             end
             % Radiation IRF
             obj.hydroForce.fDamping=zeros(nDOF,LDOF);
             irfk = obj.hydroData.hydro_coeffs.radiation_damping.impulse_response_fun.K  .*rho;
             irft = obj.hydroData.hydro_coeffs.radiation_damping.impulse_response_fun.t;
-            %obj.hydroForce.irkb=zeros(CIkt,6,lenJ);
+            %obj.hydroForce.irkb=zeros(cicLength,6,obj.dofCoupled);
```
Comment (akeeste) [inline source/objects/bodyClass.m:626]: where is ``cicLength`` defined?

**Review by akeeste — COMMENTED** [review]
What was changed in the three example .slx files?
Did we ever decide on renaming for ``momOfInertia``?

File: source/objects/waveClass.m (hunk 9 above)
Comment (kmruehl) [reply source/objects/waveClass.m:714]: `:meth:` refers to method

File: source/functions/paraview/writeParaviewBody.m [hunk 12]
```diff
@@ -1,6 +1,6 @@
-function writeParaviewBody(body, t, pos_all, bodyname, model, simdate, hspressure,wavenonlinearpressure,wavelinearpressure,pathParaviewVideo,vtkbodiesii)
+function writeParaviewBody(body, t, pos_all, bodyname, model, simdate, hspressure,wavenonlinearpressure,wavelinearpressure,paraview.path,vtkbodiesii)
```
[the remark this answers is not in the document]
Comment (kmruehl) [reply source/functions/paraview/writeParaviewBody.m:1]: It doesn't, this has been resolved on https://github.com/WEC-Sim/WEC-Sim/pull/822

File: source/functions/simulink/model/ConvolutionIntegral_interp.m [hunk 13]
```diff
@@ -1,12 +1,12 @@
-function F_FM = ConvolutionIntegral_interp(velocity, IRKB, CTTime)
+function F_FM = ConvolutionIntegral_interp(velocity, IRKB, cicTime)
```
[the remark this answers is not in the document]
Comment (kmruehl) [reply source/functions/simulink/model/ConvolutionIntegral_interp.m:1]: Yes, but let's focus on input file parameters for now. We can change internal variables later, this is just one of many that needs to be updated.

**Comment by kmruehl** [comment]
All of these comments should be addressed in https://github.com/WEC-Sim/WEC-Sim/pull/822

**Pull request merged**